### DISCUSSION TITLE
node-red-contrib-uhppoted

### DISCUSSION THREAD

**Discussion by michelebertonelettronica** [uhppoted/uhppoted#52] [General] [answered]
Hello, 

I am using :

node-red-contrib-uhppoted     
UHPPOTE Professionale Wiegand 26 Bit TCP IP Network Control Panel Control Controller per 2 porte 4 Reader

I get this error when I try to communicate with the card :

"uhppoted::Error: bind EADDRNOTAVAIL 192.168.1.220"

can you help me..?
Thank you very much

Michele Berton

**Comment by uhppoted** [comment]
Hi,

That error means you are trying to bind to an IP address that is not valid for the machine on which you are running _NodeRED_. I'm guessing that 192.168.1.220 is the controller IP address? 

The bind address should be the LAN address of the _NodeRED_ machine - although in general if you bind to `0.0.0.0:0` you should be ok (unless of course you have a more complex network setup or are perhaps running on a Mac and want to specifically bind to the Ethernet LAN and not the WiFi).

The attached screenshot is an example - you can leave the _controllers_ section blank unless you have specific network routing requirements.


![screenshot](https://github.com/uhppoted/uhppoted/assets/61560842/ba9f72f9-dc54-4d25-82f0-ba36d8775394)

**Comment by michelebertonelettronica** [comment]
Hello,



[image: image.png]


Thanks for the reply

I tried as indicated, but I get another error..

then when I press + to add a control unit comes out that is not the one
connected..,
the connected one is this:


[image: image.png]

grazie ancora

Michele

Il giorno mer 26 giu 2024 alle ore 20:00 uhppoted ***@***.***>
ha scritto:

> Hi,
>
> That error means you are trying to bind to an IP address that is not valid
> for the machine on which you are running *NodeRED*. I'm guessing that
> 192.158.1.220 is the controller IP address?
>
> The bind address should be the LAN address of the *NodeRED* machine -
> although in general if you bind to 0.0.0.0:0 you should be ok (unless of
> course you have a more complex network setup or are perhaps running on a
> Mac and want to specifically bind to the Ethernet LAN and not the WiFi).
>
> The attached screenshot is an example - you can leave the *controllers*
> section blank unless you have specific network routing requirements.
>
> screenshot.png (view on web)
> <https://github.com/uhppoted/uhppoted/assets/61560842/ba9f72f9-dc54-4d25-82f0-ba36d8775394>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/uhppoted/uhppoted/discussions/52#discussioncomment-9885868>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALMJCMUSBXFU2P7IAEQJVZLZJL6T5AVCNFSM6AAAAABJ6KUGU6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TQOBVHA3DQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
*  Michele Berton        ** +39 3515384314  *
***@***.*** ***@***.***>m*

*          soluzionielettroniche.com <http://soluzionielettroniche.com/>*

**Comment by michelebertonelettronica** [comment]
I tried as indicated, but I get another error..

![image](https://github.com/uhppoted/uhppoted/assets/47747378/997b7487-356a-4809-b8b8-044e3fd291e0)

the connected one is this:
![image (1)](https://github.com/uhppoted/uhppoted/assets/47747378/21c3affc-d815-4bdd-b0fc-09407654ccbe)

**Comment by uhppoted** [comment]
_EACESS_ means you don't have permission to send to a UDP broadcast address - which is possibly because:
- 255.255.255.255 is not the broadcast address for your network
- there is something in your configuration that is using _sendto_ instead of _broadcast_

Let's take this step by step:
1. Can you post the network configuration for the NodeRED machine - think it's [_ipconfig /all_](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/ipconfig) on _Windows_ (otherwise somewhere in the the admin tools).
2. What do you get if you run a _get-devices_ node? i.e. in your flow (or a different one) use the _get-devices_ node rather than the _get-device_ node. The difference is that _get-devices_ gets a list of all the controllers it can find using UDP broadcast whereas _get-device_ gets the information for a single controller and it  may do something different depending on the configuration.
3. Can you download the CLI from [here](https://github.com/uhppoted/uhppote-cli/releases), run the following command and post the result:
```
uhppote-cli --debug get-devices
```


re. '+'. Not sure if you figured it out - but the information shown is the placeholder information (i.e. hint text). It's slightly lighter than the real information and you just type the correct information over it

![screenshot](https://github.com/uhppoted/uhppoted/assets/61560842/7e2c126a-a415-43ed-bd36-67138549a93c)

**Comment by michelebertonelettronica** [comment, marked as answer]
HI ,
Thanks for the info..
I verified . my network is
192.168.1.254 !!
now I can communicate with the system
Thank you very much !!

I will try the functions described
Best regards and thanks again..
Michele

![image](https://github.com/uhppoted/uhppoted/assets/47747378/3975b059-9dab-4eea-b999-71ac9f700874)

**Comment by michelebertonelettronica** [comment]
Error
I verified . my network is
192.168.1.255 !!

**Reply by uhppoted** [reply]
I was *just* about to comment that 192.168.1.254 is very unusual :-) .. but **great**, glad you're up and running!

The other steps were just diagnostics to help me figure out what was happening on your network so there's no need to do them (except of course for your own interest).

**Comment by uhppoted** [comment]
On Mon, 29 Dec 2025 at 03:09, Peter ***@***.***> wrote:

> Thanks for the above discussion. Can I ask a follow-on question - does the
> "listen" node in uhppoted work? Although I can poll the wiegand RFID access
> controller from the nodes "get-event" and 'get-status", I can't get any
> response when I set up a listen node with the same configyration.

Hi,

I can't find this question anywhere in the *uhppoted* github repo - have
you perhaps deleted it?

**Comment by peterw721** [comment]
Hi,
Yep after a couple months of frustration i see I was incorrectly setting uhppoted to listen on the controller's IP address instead of the Pi's Ip address. Thanks for the good info in the thread!
________________________________

**Reply by uhppoted** [reply]
I saw that :-) .. you aren't the first person to do that and almost certainly not the last!! Glad you got it sorted though!

I'll think about adding something to the FAQ.

**Comment by michelebertonelettronica** [comment]
Hi Peter, I haven't had a chance to do any more tests. I needed to run some
tests with Nodered for access control.
I haven't done any more in-depth tests.
If I need any more help, I hope to have your help again.
Thanks so much, and sorry for my English. I often use the translator.

Michele

Il giorno mar 30 dic 2025 alle ore 11:44 Peter ***@***.***>
ha scritto:

> Hi,
> Yep after a couple months of frustration i see I was incorrectly setting
> uhppoted to listen on the controller's IP address instead of the Pi's Ip
> address. Thanks for the good info in the thread!
> ________________________________
> From: uhppoted ***@***.***>
> Sent: 30 December 2025 06:30
> To: uhppoted/uhppoted ***@***.***>
> Cc: Peter ***@***.***>; Comment ***@***.***>
> Subject: Re: [uhppoted/uhppoted] node-red-contrib-uhppoted (Discussion
> #52)
>
>
> Hi,
>
> I can't find this question anywhere in the *uhppoted* github repo - have
> you perhaps deleted it?
>
> On Mon, 29 Dec 2025 at 03:09, Peter ***@***.***> wrote:
>
> > Thanks for the above discussion. Can I ask a follow-on question - does
> the
> > "listen" node in uhppoted work? Although I can poll the wiegand RFID
> access
> > controller from the nodes "get-event" and 'get-status", I can't get any
> > response when I set up a listen node with the same configyration.
> >
> > debug_1.jpg (view on web)
> > <
> https://github.com/user-attachments/assets/c42d82ee-24d5-4d0f-907f-be97409697c2>
>
> > debug_2.png (view on web)
> > <
> https://github.com/user-attachments/assets/ebbb4f0e-924f-4b07-a522-40cdebb62c0b>
>
> >
> > Any feedback much appreciated!
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/uhppoted/uhppoted/discussions/52#discussioncomment-15363882>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AOVVQCWOAJT6VGURNNMK6JD4EEDWHAVCNFSM6AAAAACQHHTF7WVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTKMZWGM4DQMQ>
>
> > .
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/uhppoted/uhppoted/discussions/52#discussioncomment-15368907>,
> or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/A2JJH7NWGKQMRJQLCM74HWD4EFQJ7AVCNFSM6AAAAACQHHTF7WVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTKMZWHA4TANY>.
>
> You are receiving this because you commented.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/uhppoted/uhppoted/discussions/52#discussioncomment-15374213>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALMJCMQRF2NSXPH2ASVL6H34EJJRNAVCNFSM6AAAAACQHHTF7WVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTKMZXGQZDCMY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
*  Michele Berton        ** +39 3515384314  *
***@***.*** ***@***.***>m*

*          soluzionielettroniche.com <http://soluzionielettroniche.com/>*

**Comment by peterw721** [comment]
Thank you Michele 🙂

https://www.youtube.com/watch?v=d8-XKxpn9YA

________________________________

**Reply by uhppoted** [reply]
Oh! That is **SO** cool! I shall put a link to it in the README ...

**Comment by peterw721** [comment]
awesome - thanks Michele 
happy New Year & All the B for 2026!!
________________________________

**Reply by uhppoted** [reply]
And all the best for 2026 to you (and of course the railroad) too :-).
